### PR TITLE
Use iam to get tls config

### PR DESCRIPTION
**Pull request by MykolaSolyanko**

### CODE REVIEW

File: iamclient/iamclient.go [hunk 1]
```diff
@@ -273,6 +272,62 @@ func (client *Client) Close() (err error) {
  * Private
  **********************************************************************************************************************/
 
+func createPublicConnection(
+	serverURL string, cryptocontext *cryptutils.CryptoContext,
+	insecure bool) (connection *grpc.ClientConn, err error) {
```
Comment (Eabramov84) [inline iamclient/iamclient.go:277]: In this case, Please use next long line formatting
func createPublicConnection(serverURL string, cryptocontext *cryptutils.CryptoContext,insecure bool) (
    connection *grpc.ClientConn, err error) {

File: iamclient/iamclient.go [hunk 2]
```diff
@@ -273,6 +272,62 @@ func (client *Client) Close() (err error) {
...
+	}
+
+	if connection, err = grpc.DialContext(ctx, serverURL, secureOpt, grpc.WithBlock()); err != nil {
+		return nil, aoserrors.Wrap(err)
+	}
+
+	return connection, nil
+}
+
+func (client *Client) createProtectedConnection(
```
Comment (Eabramov84) [inline iamclient/iamclient.go:301]: the same

File: iamclient/iamclient.go [hunk 3]
```diff
@@ -273,6 +272,62 @@ func (client *Client) Close() (err error) {
...
+	config *config.Config, cryptocontext *cryptutils.CryptoContext,
+	insecure bool) (connection *grpc.ClientConn, err error) {
+	var secureOpt grpc.DialOption
+
+	ctx, cancel := context.WithTimeout(context.Background(), iamRequestTimeout)
```
Comment (Eabramov84) [inline iamclient/iamclient.go:306]: please create context before use () move to line 323

File: iamclient/iamclient_test.go [hunk 4]
```diff
@@ -517,31 +570,31 @@ func (server *testServer) UnregisterService(
...
-func (server *testServer) GetPermissions(
+func (server *testPublicServer) GetPermissions(
 	ctx context.Context, req *pb.PermissionsRequest) (rsp *pb.PermissionsResponse, err error) {
 	rsp = &pb.PermissionsResponse{}
 
-	funcServersPermissions, ok := server.permissionsCache[req.Secret]
-	if !ok {
-		return rsp, aoserrors.New("secret not found")
-	}
+	// funcServersPermissions, ok := server.permissionsCache[req.Secret]
```
Comment (Eabramov84) [inline iamclient/iamclient_test.go:577]: why this code commented?

File: communicationmanager.go [hunk 5]
```diff
@@ -232,7 +239,7 @@ func (cm *communicationManager) close() {
 		cm.alerts.Close()
 	}
 
-	// Close crypto context
+	// Close crypto handler
```
Comment (Eabramov84) [inline communicationmanager.go:243]: please move closing of crypto handler and  cryptoContext after closing qmqp

File: smcontroller/smcontroller.go [hunk 6]
```diff
@@ -55,7 +55,7 @@ type Controller struct {
 	clientsWG        sync.WaitGroup
 	readyWG          sync.WaitGroup
 	clients          map[string]*smClient
-	context          context.Context
+	ctx              context.Context // nolint:containedctx
```
Comment (Eabramov84) [inline smcontroller/smcontroller.go:58]: does it necessary to rename variable?

File: cmserver/cmserver.go [hunk 7]
```diff
@@ -103,7 +103,7 @@ type CMServer struct {
 	sync.Mutex
 }
 
-// CertificateProvider certificate and key provider interface
+// CertificateProvider certificate and key provider interface.
```
Comment (Eabramov84) [inline cmserver/cmserver.go:106]: please fix it in originla commits

File: communicationmanager.go [hunk 8]
```diff
@@ -175,7 +175,8 @@ func newCommunicationManager(cfg *config.Config) (cm *communicationManager, err
 	}
 
 	// Create SM controller
-	if cm.smController, err = smcontroller.New(cfg, cm.amqp, cm.alerts, cm.monitor, cm.fileServer, cm.iam, cm.cryptoContext, false); err != nil {
+	if cm.smController, err = smcontroller.New(cfg, cm.amqp, cm.alerts, cm.monitor,
```
Comment (Eabramov84) [inline communicationmanager.go:178]: the same

File: iamclient/iamclient.go (hunk 1 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient.go]: Done

File: iamclient/iamclient.go (hunk 2 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient.go:300]: Done

File: iamclient/iamclient.go (hunk 3 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient.go]: Done

File: iamclient/iamclient.go [hunk 9]
```diff
@@ -273,6 +272,62 @@ func (client *Client) Close() (err error) {
...
+		certURL, keyURL, err := client.GetCertificate(config.CertStorage, nil, "")
+		if err != nil {
+			return nil, aoserrors.Wrap(err)
+		}
+
+		tlsConfig, err := cryptocontext.GetClientMutualTLSConfig(certURL, keyURL)
+		if err != nil {
+			return nil, aoserrors.Wrap(err)
+		}
+		secureOpt = grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig))
```
[the remark this answers is not in the document]
Comment (MykolaSolyanko) [reply iamclient/iamclient.go:318]: Done

File: iamclient/iamclient_test.go (hunk 4 above)
Comment (MykolaSolyanko) [reply iamclient/iamclient_test.go]: this method is not used. left after testing

File: communicationmanager.go (hunk 5 above)
Comment (MykolaSolyanko) [reply communicationmanager.go]: Done

File: smcontroller/smcontroller.go (hunk 6 above)
Comment (MykolaSolyanko) [reply smcontroller/smcontroller.go]: revert

File: communicationmanager.go (hunk 8 above)
Comment (MykolaSolyanko) [reply communicationmanager.go:178]: Done

**Pull request merged**